### ISSUE TITLE
Language model pre-training hyper-params, batch sizes

### ISSUE THREAD

**Issue by tpietruszka** [tpietruszka/ulmfit_attention#3]
The reason I am curious, is I've not had luck with ULMFiT with lower batch sizes.  It seems the lower the batch size the worse the results (no matter how long I train).  Which sucks since I actually have some 1070's sitting around doing nothing.  Instead, I am having to train on a p3.2xlarge up in amazon.

Basically I've seen a linear relationship between batch size and performance.  Maybe is just because some other hyperparameter needs to be tuned to deal with lower batch sizes.

_Originally posted by @dalupus in https://github.com/tpietruszka/ulmfit_attention/issues/1#issuecomment-618109689_

**Comment by tpietruszka** [comment]
Interesting. I think my results did not change too much, except if I had to lower BPTT. I did adjust the learning rate AFAIR.

Seems like I dropped the ball here, regarding reproducibility. Although I still hope I will be able to dig up the notebook I used.

Just to make sure - was the relationship between bs and performance you saw:

1. during language model pre-training/fine-tuning, in terms of perplexity
2. during language model pre-training/fine-tuning, in terms of performance of the downstream classifier
3. during classifier training

or perhaps all of the above?

Did you adjust the learning rate in any way?

**Comment by dalupus** [comment]
I saw a performance decrease across the board.  I did adjust the learning rate, however, I did not touch any of the other parameters.  I have been thinking that it might be that the momentum needs to be adjusted for smaller batch sizes.

Note that this was with my review dataset and not the IMDB one.  It could be IMDB is simply to easy of a problem for these types of things to matter.
